### PR TITLE
[Foundation] Add bindings for Xcode 13 rc 1.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: tests/monotouch-test/monotouch-test.csproj [hunk 1]
```diff
@@ -394,7 +394,7 @@
     <Exec Command="xcrun -sdk $(_SmeltingSdk) metal -c @(CustomMetalSmeltingInput) -o $(IntermediateOutputPath)\fragmentShader.air $(_SmeltingMinOS)" EnvironmentVariables="DEVELOPER_DIR=$(_SdkDevPath)" />
     <Exec Command="xcrun -sdk $(_SmeltingSdk) metallib $(IntermediateOutputPath)\fragmentShader.air -o $(AppBundleDir)\fragmentShader.metallib" EnvironmentVariables="DEVELOPER_DIR=$(_SdkDevPath)" />
   </Target>
-  <Target Name="BeforeBuild" Inputs="@(GeneratedTestInput)" Outputs="@(GeneratedTestOutput)" DependsOnTargets="CustomMetalSmelting" >
```
Comment (chamons) [inline tests/monotouch-test/monotouch-test.csproj:397]: Seems unrelated to PR?

**Review by chamons — APPROVED** [review]
Looks reasonable. There are a lot of tests, but I'm not sure enough given how much manual bindings/wraps there were.

I want @rolfbjarne 's blessing before this goes in.

Good work, this was messy.

File: tests/monotouch-test/Foundation/NSArray1Test.cs [hunk 2]
```diff
@@ -93,5 +95,28 @@ public void FromNSObjectsNullTest ()
...
+
+		[Test]
+		public void GetDifferenceFromArrayTest ()
+		{
+			TestRuntime.AssertXcodeVersion (13,0);
+			using var str1 = (NSString) "1";
+			using var str2 = (NSString) "1";
+			using var str3 = (NSString) "1";
+			
+			using var array1 = NSArray.FromObjects (str1, str2 );
```
Comment (tj-devel709) [inline tests/monotouch-test/Foundation/NSArray1Test.cs:107]:
```suggestion
			using var array1 = NSArray.FromObjects (str1, str2);
```nit

File: tests/monotouch-test/Foundation/NSOrderedCollectionChange1Test.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+			Assert.AreEqual ((nuint)0, change.Index, "Index");
+		}
+
+		[Test]
+		public void ChangeWithObjectWithAssociatedIndexTest ()
+		{
+			TestRuntime.AssertXcodeVersion (13,0);
+
+			var str = new NSString ("Test");
+			var change = NSOrderedCollectionChange<NSString>.ChangeWithObject(str, NSCollectionChangeType.Insert, 0, 1);
```
Comment (tj-devel709) [inline tests/monotouch-test/Foundation/NSOrderedCollectionChange1Test.cs:30]:
```suggestion
			var change = NSOrderedCollectionChange<NSString>.ChangeWithObject (str, NSCollectionChangeType.Insert, 0, 1);
```nit

**Review by tj-devel709 — COMMENTED** [review]
Just some small things I noticed!

File: src/generator.cs [hunk 4]
```diff
@@ -1785,7 +1785,7 @@ public TrampolineInfo MakeTrampoline (Type t)
 		}
 
 		var rt = mi.ReturnType;
-		var rts = IsNativeEnum (rt) ? "var" : rt.ToString ();
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply src/generator.cs:1788]: Yes, without this fix you will get an issue with the trampolines that use generics because you get the class name like NSArra'1 rather than NSArray<String>

File: src/foundation.cs [hunk 5]
```diff
@@ -3569,7 +3668,10 @@ interface NSMutableArray {
 		[iOS (8,0), Mac(10,10)]
 		[Static, Export ("arrayWithContentsOfURL:")]
 		NSMutableArray FromUrl (NSUrl url);
-
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply src/foundation.cs:3572]: It was not removed. Someone left a tab and is added in the diff.

File: src/generator.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (chamons) [reply src/generator.cs:1788]: Ideally this would be in a separate PR to make blaming it in the future more obvious.

Consider calling it out with explaination in the PR/commit description for possible future us.

**Review by dalexsoto — APPROVED** [review]
Nice! 👍++ for the generator fix and the tests!

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Comment by dalexsoto** [comment]
@mandel-macaque is this still a thing? we may want to land it before @chamons converts attributes on Foundation

File: src/generator.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply src/generator.cs:1795]: This can be a diff fix. The issue where is that the rt.ToString() ois doing the wrong thing and is generating a non-compilable name. I'll add this as a separate branch with a decent explanation.
